app/src/main/java/com/codesoom/assignment/repository/TaskRepository.java

```diff
@@ -0,0 +1,88 @@
+package com.codesoom.assignment.repository;
+
+import com.codesoom.assignment.models.Task;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Task Entity의 데이터를 보관하고 이 Entity의 CRUD와 관련된 로직을 담당합니다.
+ */
+@SpringBootApplication(scanBasePackages={"com.codesoom.assignment.repository"})
+public class TaskRepository {
+    private List<Task> tasks = new ArrayList<>();
+    private Long newId = 0L;
+
+    /**
+     * 저장된 할 일들(tasks)을 모두 반환합니다.
+     *
+     * @return 모든 할 일들(tasks)
+     */
+    public List<Task> findAll(){
+        return tasks;
+    }
+
+    /**
+     * 새로운 할 일(task)을 생성합니다.
+     *
+     * @param : 할 일(task)의 내용
+     * @return : 저장된 할 일(task)을 반환합니다.
+     */
+    public Task create(Task task){
+        task.setId(generateId());
+        tasks.add(task);
+        return task;
+    }
+
+    /**
+     * 식별자(id)에 해당하는 할 일(task)을 수정한다.
+     *
+     * @param : 수정하려는 할 일(task)의 식별자(id)
+     * @param : task(할 일)의 내용
+     * @return 수정한 할 일(task)의 내용울 반환합니다.
+     */
```

Comment:
```suggestion
    /**
     * id에 해당하는 할 일의 내용을 수정하고, 수정된 할 일을 리턴합니다.
     *
     * @param id 할 일의 식별자
     * @param source 업데이트 정보
     * @return 수정된 할 일
     */
```

summary(첫째 줄)에서 제일 중요한 것은 무엇을 리턴하는지라 할 수 있어요. 이 메소드는 단순하지만, 만약 처음 보는 어떤 메소드 `updateTaskWithHash` 가 길이가 100줄이고, 내용도 복잡해서 이해하기 까다롭다고 상상해 보세요. 

```java
public Task updateTaskWithHash(String sha256Hash, HashEncoder encoder, Task task) {
  // 100줄짜리 코드
  return...
}
```

파악하는 데에만 아주 오랜 시간이 걸릴 수 있어요.

그런데 그 메소드에 javadoc이 있고 다음과 같은 내용이라 생각해 봅시다.
```java
/**
 * 주어진 hash값에 해당하는 할 일의 내용을 업데이트하고 리턴합니다.
 * 
 * @param hash 할 일의 해시값을 16진수로 표현한 문자열
 * @param encoder 해시값 인코더
 * @param task 할 일 인스턴스
 * @return 업데이트 완료된 할 일 인스턴스
 * @throws TaskNotFoundException 해시값으로 할 일을 찾지 못한 경우
 * @throws InvalidHashEncoderException 적합하지 않은 해시 인코더가 주어진 경우
 */
public Task updateTaskWithHash(String sha256Hash, HashEncoder encoder, Task task) {
```

어떤가요? 메소드를 보지도 않았는데도 대충 어떻게 사용할 수 있고, 어떤 점을 주의해야 하는지 큰 힌트가 되죠.
설령 이해가 잘 안 간다 하더라도 `summary`를 읽고 "업데이트된 할 일이 리턴되는 건가 보다" 라는 정도까지는 생각할 수 있습니다. 어떻게든 이해의 실마리가 되는 거죠.